### DISCUSSION TITLE
Flashing MicroPython on ESP32 and ESP8266 using the web browser

### DISCUSSION THREAD

**Discussion by rafaelaroca** [BIPES/BIPES#81] [Announcements]
It is now possible to flash ESP32 or ESP8266 directly from the browser! No esptool install needed anymore :-)

Just go to:

https://bipes.net.br/flash/esp-web-tools/

Plug the board on the USB port, click Connect, and install MicroPython on your board. 

This allows a quick setup to get started with MicroPython and BIPES.

**Comment by pmalta** [comment]
How long could takes to inastall MicroPython?
I have 2 NodeMCU "Amica" based on ESP12E/F and Silabs CP2102 USB-Uart. 
I am using, Win10, CP2102 is installed and boards are recognized.
None of them installed MicroPython using browser. I tested in Opera & Chrome
I made all steps and browser remains stucked in a small window with message: "Installing MicroPython 1.17 ... Preparing installation"
![image](https://user-images.githubusercontent.com/14170683/155978757-659996b5-a3b5-4a54-bcf4-8223b6ad9016.png)

**Comment by rafaelaroca** [comment]
Hello @pmalta! Pleased to see you here!
We have seen this situation with Amica boards, and some were solved by holding the BOOT button of the board while "preparing installation" is shown.

best regards. good to talk to you!
Rafael

**Reply by pmalta** [reply]
Hi @rafaelaroca, I guess my 2 "Amica" boards are demage, they was already used in other projects.
I repeated procedure with a "Heltec ESP32 with LoRa" and procedure executed as expected.
I'll look for a simple test to these ESP 8266 NodeMCU to confirm they are OK or not. 
I'll return to you if they are OK to help you to debug flash procedure for them.
If they aren't OK I think I have some ESP12F new from the box that I can use to save these nodeMCU.
Regards

**Comment by rafaelaroca** [comment]
Nice! By the way, have you seen our recent ebooks related to BIPES? They are good starting points for BIPES with ESP32 and ESP8266:

https://bipes.net.br/wp/book-livro/

and

https://obsat.org.br/livro

best regards!
Rafael

**Reply by pmalta** [reply]
I already downloaded your team book about BIPES.
I started to use it this _Carnival_ ... that's why I faced this problem with the nodeMCU board.
Regards

**Comment by pmalta** [comment]
I guess my ESP8266 has some problem ...
I follwed a procedure from nodeMCU to load original firmware to boards.
One of them didn't worked, the other worked and cheking in Arduino terminal I got
_NodeMCU 3.0.0.0 built on nodemcu-build.com provided by frightanic.com
	branch: release
	commit: f25dc56d3c6213b8ac7ce46d1293466137746eae
	release: 
	release DTS: 202112300746
	SSL: false
	build type: integer
	LFS: 0x0 bytes total capacity
	modules: file,gpio,net,node,tmr,uart,wifi
 build 2022-02-28 15:58 powered by Lua 5.1.4 on SDK 3.0.1-dev(fce080e)
cannot open init.lua:_ 
Even after this procedure the MicroPython web flashing didn't worked for the board
I'll try to replace ESP12F on these boards

**Reply by pmalta** [reply]
Apparently I got MicroPython in one of the boards.
I used NodeMCU pyFlasher as recommmended in NodeMCU documentation (https://nodemcu.readthedocs.io/en/latest/flash/) and flashed MicroPython 1.17 from MicroPython repository (https://micropython.org/download/esp8266/).
I checked with BIPES console and board is apparently OK.
Now I have 1 ESP8266 and 1 ESP32 to work with BIPES.
When I replace the ESP12F in the other board I'll check again the MicroPython flashing using browser

**Comment by abishekmuthian** [comment]
Head's up;  esp-web-tools greater than 6.1.1 doesn't work with ESP82666 as it's dependency removed the required header patching(https://github.com/NabuCasa/esp-web-flasher/issues/103).

I'm commenting here because I used your tool for figuring this out i.e. Your application uses version 6.1.1 and can flash micropython successfully to ESP8266 where as later versions results in reset loop(not because of power issue).

If you have expertise in flashing modes for ESP8266 then your input can be valuable in the aforementioned issue.

**Reply by rafaelaroca** [reply]
Hello @abishekmuthian 
Thank you for the warning. Flashing ESP8266 with MicroPython from the web is an important tool for us. We will keep this in mind for future updates of esp-web-flasher.

thanks!
Rafael

**Reply by abishekmuthian** [reply]
Yes, Web flashing is great for reducing friction but unfortunately the header patching routines seem to have removed hastly due to lack of participation of those who use ESP8266 boards and in favour or ESP32 boards.

**Comment by solarkraft** [comment]
I wish the MicroPython version wasn't outdated. I don't blame you for not keeping it up to date, I blame there seemingly not being a way to _always_ obtain the most recent version. 
I asked the MicroPython people [a question about getting a stable link to the latest release](https://github.com/orgs/micropython/discussions/10519).
